exec/java-exec/src/test/java/org/apache/drill/TestSelectWithFullNameIdentifiers.java

```diff
@@ -0,0 +1,180 @@
+      // Query with using scalar function, WHERE statement, ORDER BY statement
+    testBuilder()
+        .sqlQuery("SELECT cp.`employee.json`.employee_id, UPPER(cp.`employee.json`.full_name) full_name " +
+            "FROM cp.`employee.json` " +
+            "WHERE cp.`employee.json`.department_id = 4 " +
+            "ORDER BY cp.`employee.json`.full_name")
+        .unOrdered()
+        .baselineColumns("employee_id", "full_name")
+        .baselineValues(43L, "JUANITA SHARP")
+        .baselineValues(44L, "SANDRA BRUNNER")
+        .build()
+        .run();
+  }
+
+  @Test
+  @Ignore
+  public void testQueryWithStar() throws Exception {
+    // Query with using full schema name for star identifier
+    // TODO: Can be used after resolving "CALCITE-1323: Wrong prefix number in DelegatingScope.fullyQualify()"
+    testBuilder()
+        .sqlQuery("SELECT cp.`department.json`.* FROM cp.`department.json` LIMIT 1")
+        .unOrdered()
+        .baselineColumns("department_id", "department_description")
+        .baselineValues(1L, "HQ General Management")
+        .build()
+        .run();
+  }
+
+  @Test
+  @Ignore
+  public void testQueryWithAggregation() throws Exception {
+    // Query with aggregation
+    // TODO: Can be used after resolving "DRILL-3993: Rebase Drill on Calcite master branch"
+    // Fixed in "CALCITE-881: Allow schema.table.column references in GROUP BY".
+    testBuilder()
+        .sqlQuery("SELECT cp.`employee.json`.position_title, " +
+            "COUNT(cp.`employee.json`.employee_id) employee_number " +
+            "FROM cp.`employee.json` " +
+            "WHERE cp.`employee.json`.position_title = 'Store Permanent Stocker' " +
+            "GROUP BY cp.`employee.json`.position_title")
+        .unOrdered()
+        .baselineColumns("position_title", "employee_number")
+        .baselineValues("Store Permanent Stocker", 222L)
+        .build()
+        .run();
+
+  }
+
+  @Test
+  public void testLeftJoin() throws Exception {
+    // Query with left join (with different schema-qualified tables)
+    testBuilder()
+        .sqlQuery("SELECT cp.`employee.json`.employee_id, cp.`employee.json`.full_name, dfs_test.tmp.`department`.department_id " +
+            "FROM cp.`employee.json` LEFT JOIN dfs_test.tmp.`department` " +
+            "ON cp.`employee.json`.department_id = dfs_test.tmp.`department`.department_id " +
+            "WHERE dfs_test.tmp.`department`.department_description = 'HQ Marketing' " +
+            "ORDER BY cp.`employee.json`.full_name")
+        .unOrdered()
+        .baselineColumns("employee_id", "full_name", "department_id")
+        .baselineValues(36L, "Donna Arnold", 3L)
+        .baselineValues(42L, "Doris Carter", 3L)
+        .baselineValues(41L, "Howard Bechard", 3L)
+        .baselineValues(7L, "Rebecca Kanagaki", 3L)
+        .build()
+        .run();
+  }
+
+  @Test
+  @Ignore
+  public void testNestedQueryInWhereStatement() throws Exception {
```

Comment:
I'm not fully convinced that the current patch is ready to merge, before we could resolve the case where full-schema qualified names appear in a subquery.  

With your patch, looks like we can only allow fully-schema qualified name in a top-level query, which is kind of restricted use case. It's hard to tell Drill user that they only can use one special case, but not for other cases.

I would suggest we look into the Calcite in more detail, to understand the problem more.